src/assets/toolkit/scripts/lib/component/float-label.js

```diff
@@ -2,26 +2,32 @@
 
 export class FloatLabel {
   /**
+   * @see https://mathiasbynens.be/notes/oninput
    * @param {DOM} element
    * @param {Object} options
    * @param {String} options.inputElement - Reference to input within element.
-   * @param {String} options.eventName - Event that will trigger updates.
    * @param {String} options.className - Class applied to `element` when `inputElement` is empty.
    */
   constructor(element, {
     inputElement = element.querySelector('input, textarea'),
-    eventName = 'input',
     className = 'is-empty'
   } = {}) {
     // Assign properties
     Object.assign(this, {
       element,
       inputElement,
-      eventName,
       className
     });
-    // Listen for event and update
-    this.element.addEventListener(this.eventName, this.update.bind(this));
+    // Fallback event handler
+    var keyUpHandler = () => this.update();
+    // Ideal event handler, will remove the other one if supported
+    var inputHandler = () => {
+      this.element.removeEventListener('keyup', keyUpHandler);
```

Comment:
I wondered the same thing. Potentially, I could do something like this...

```js
// Create the generic, bound event handler
var eventHandler = () => this.update();
// Create a bound event handler for the first input event
var inputCheck = () => {
  // Remove the keyup handler
  this.element.removeEventListener('keyup', eventHandler);
  // Remove this check handler
  this.element.removeEventListener('input', inputCheck);
  // Add the generic event handler to this event
  this.element.addEventListener('input', eventHandler);
  // Fire the update
  this.update();
};
// Attach events
this.element.addEventListener('input', inputCheck);
this.element.addEventListener('keyup', eventHandler);
```

But that seemed like extra complexity, and Mathias's stuff (who was [the source](https://mathiasbynens.be/notes/oninput) for this) tends to be pretty performant.

I'd be interested in @erikjung's opinion here as well.